packages/database/database.go

```diff
@@ -1,134 +1,65 @@
 package database
 
 import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"runtime"
 	"sync"
-	"time"
 
-	"github.com/dgraph-io/badger"
+	"github.com/dgraph-io/badger/v2"
+	"github.com/iotaledger/goshimmer/packages/parameter"
+	"github.com/iotaledger/hive.go/database"
 )
 
 var (
-	ErrKeyNotFound = badger.ErrKeyNotFound
-
-	dbMap = make(map[string]*prefixDb)
-	mu    sync.Mutex
+	instance *badger.DB
+	once     sync.Once
 )
 
```

Comment:
Add something like the following, so that it is no longer necessary to also include `hive.go/database` each time the database is used in goshimmer:
```Go
type Database = database.Database

func Get(dbPrefix byte, optionalBadger ...*badger.DB) (Database, error) {
	return database.Get(dbPrefix, optionalBadger...)
}
```

Reply:
This doesn't work unfortunately really, because I still need to import the `hive.go` database package as I can't type alias the `ErrKeyNotFound` which is virtually used by every file which also initiates a database.

Reply:
I don't fully understand, shouldn't the following be sufficient to not import `hive.go/database` throughout goshimmer:
```go
var ErrKeyNotFound = database.ErrKeyNotFound
type (
	Database     = database.Database
	Entry        = database.Entry
	KeyOnlyEntry = database.KeyOnlyEntry
)
func Get(dbPrefix byte, optionalBadger ...*badger.DB) (Database, error) {
	return database.Get(dbPrefix, optionalBadger...)
}
```

Reply:
You're obviously right. I must have tried to use `type ErrKeyNotFound` 🤦‍♂